src/ucp/core/ucp_mm.h

```diff
@@ -41,9 +41,25 @@ typedef struct ucp_mem {
  */
 typedef struct ucp_mem_desc {
     ucp_mem_h                     memh;
+    ucp_mem_h                     meta_memh;
+    void                          *ptr;
+    void                          *meta_ptr;
+    ucs_memory_type_t             mem_type;
+    volatile uint32_t             count;
```

Comment:
1. no need meta_memh
2. no need meta_ptr. desc itself is the ptr.
3. no need memtype and count